tests/ltp/umount02.patch

```diff
@@ -0,0 +1,81 @@
+diff --git a/testcases/kernel/syscalls/umount/umount02.c b/testcases/kernel/syscalls/umount/umount02.c
+index cd67aa6d2..8822eab89 100644
+--- a/testcases/kernel/syscalls/umount/umount02.c
++++ b/testcases/kernel/syscalls/umount/umount02.c
+@@ -20,6 +20,11 @@
+ #include "tst_test.h"
+ 
+ #define MNTPOINT	"mntpoint"
++const char* src  = "/dev/vda";
++const char* trgt = MNTPOINT; 
++const char* type = "ext4";
++const unsigned long mntflags = 0;
++const char* opts = "mode=0777";  
+ 
+ static char long_path[PATH_MAX + 2];
+ static int mount_flag;
+@@ -31,7 +36,7 @@ static struct tcase {
+ 	int exp_errno;
+ } tcases[] = {
+ 	{"Already mounted/busy", MNTPOINT, EBUSY},
+-	{"Invalid address", NULL, EFAULT},
++/*	{"Invalid address", NULL, EFAULT}, This test is commented because in SGX accessing NULL results in enclave exit */
```

Comment:
Is that a bug in SGX-LKL? Should it be recorded in an issue?

Reply:
Yes  will raise this as an issue..  in umount02 when it test umount with null, it causes enclave to abort. Normally application is expected to get EFAULT. But the back trace is showing that in lkl it calls oe_abort when address is invalid.
 
#0  lkl_access_ok (addr=0, size=1) at lkl/setup.c:1499
#1  0x00007fff0010aaff in __lkl_access_ok (size=<optimized out>, addr=<optimized out>) at ./arch/lkl/include/asm/uaccess.h:14
#2  strncpy_from_user (count=<optimized out>, src=<optimized out>, dst=<optimized out>) at ./include/asm-generic/uaccess.h:233
#3  getname_flags (filename=0x0, flags=0, empty=0x0) at fs/namei.c:149
#4  0x00007fff0010ad2d in getname (filename=<optimized out>) at fs/namei.c:209
#5  0x00007fff0010da1d in user_path_mountpoint_at (dfd=0, name=<optimized out>, flags=1049890816, path=0x7fff3e941020)
    at fs/namei.c:2745
#6  0x00007fff0012003d in ksys_umount (name=0x7fff3e941000 " \020\224>\377\177", flags=0) at fs/namespace.c:1680
#7  0x00007fff00120479 in __do_sys_umount (flags=<optimized out>, name=<optimized out>) at fs/namespace.c:1706
#8  __se_sys_umount (name=<optimized out>, flags=<optimized out>) at fs/namespace.c:1704
#9  0x00007fff00088076 in run_syscall (params=<optimized out>, no=<optimized out>) at arch/lkl/kernel/syscalls.c:44
#10 lkl_syscall (no=140734243278880, params=0x0) at arch/lkl/kernel/syscalls.c:183
#11 0x00007fff0054901a in __filter_syscall2 (n=39, a1=0, a2=0) at ./src/internal/syscall.h:161
#12 0x00007fff0054924a in umount (special=0x0) at src/linux/mount.c:11
    if (!ret)
    {
        sgxlkl_fail("lkl_access_check failed: %p\n", (void*)addr);
    }
    return ret;